pyproject.toml

```diff
@@ -29,6 +29,7 @@ dependencies = [
   "alembic>=1.12.0",
   "typing-extensions>=4.0.0",
   "greenlet; sys_platform == \"darwin\"",
+  "typing-extensions; python_version < \"3.11\""
```

Comment:
IDT this actually constrains anything b/c its also declared as a dependency on line 30.

```
(.venv) peter@pop-os:~/PycharmProjects/aa-312-test$ python --version
Python 3.12.2
(.venv) peter@pop-os:~/PycharmProjects/aa-312-test$ python -m pip freeze
(.venv) peter@pop-os:~/PycharmProjects/aa-312-test$ python -m pip install git+https://github.com/jolt-org/advanced-alchemy.git@self-typing
Collecting git+https://github.com/jolt-org/advanced-alchemy.git@self-typing
--snipped--
(.venv) peter@pop-os:~/PycharmProjects/aa-312-test$ python -m pip freeze
advanced_alchemy @ git+https://github.com/jolt-org/advanced-alchemy.git@ee7bd61a0901420f603faf9b5d4cd7f1eae6fdd1
alembic==1.13.1
greenlet==3.0.3
Mako==1.3.2
MarkupSafe==2.1.5
SQLAlchemy==2.0.29
typing_extensions==4.10.0
(.venv) peter@pop-os:~/PycharmProjects/aa-312-test$ 
```

Reply:
Oops.  I meant to actually remove the other one :)

Reply:
And actually, I'm going to just close this one.  I see that we are using `typing_extensions` in other places as well.